### PR TITLE
Updater improvements to send option component information

### PR DESCRIPTION
**Pull request by infamy**
Updater improvements to send option component information, https://github.com/home-assistant/home-assistant/pull/7720

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

### CODE REVIEW

File: source/_components/updater.markdown [hunk 1]
```diff
@@ -23,6 +23,22 @@ updater:
 
 If you choose not to share any information when checking for updates, you can add `reporting: False`.
 
+The other option, is to share a little bit more information. You can enable sharing of component information. To do this you have to add `share_components_info: True`. 
+
+The data sent back is fairly limitied:
+```json
+"components": [
```
Comment (Landrash) [inline source/_components/updater.markdown]: Extra [ in example code

File: source/_components/updater.markdown [hunk 2]
```diff
@@ -23,6 +23,22 @@ updater:
...
+    "apcupsd",
+    "api",
+    "automation",
+    "binary_sensor",
+    "binary_sensor.zwave",
+    "camera",
+    "camera.uvc",
+    "config",
+    "config.core"...
```
Comment (Landrash) [inline source/_components/updater.markdown]: Remove dot's since their not valid in the example.

File: source/_components/updater.markdown [hunk 3]
```diff
@@ -23,6 +23,22 @@ updater:
...
+```
+
```
Comment (Landrash) [inline source/_components/updater.markdown]: Documenatatio for configuration variables will need to be added.

**Comment by andrey-git** [comment]
Since this is new *and* OptIn, how about making the reported data aggregated-public? I.e. publish the count for each component?

**Comment by balloob** [comment]
@andrey-git that is a good idea. Looping in @robbiet480 on how we should do that. I guess the easiest for now is to just generate a report every month or something.

**Pull request merged**